scripts/deploy.ts

```diff
@@ -110,6 +178,90 @@ async function main() {
   )
 }
 
+const legacyContractAddress = '0x9e7A15E77e5E4f536b8215aaF778e786005D0f8d'
+
+async function downloadData(
+  factory: OBSSStorage__factory,
+  provider: JsonRpcProvider,
+  signer: Signer
+) {
+  const legacyContract = factory
+    .attach(legacyContractAddress)
+    .connect(provider)
+    .connect(signer)
+
+  console.log(legacyContract.address)
+  const totalFeeds = await legacyContract.lastFeedId()
+  console.log(`Total feeds count: ${totalFeeds.toNumber()}`)
+  const legacyPosts: OBSSStorage.LegacyPostStruct[] = []
+  const legacyReactions: OBSSStorage.LegacyReactionStruct[] = []
+  for (let i = 0; i < 10; i++) {
+    const postsInFeed = await legacyContract.lastFeedPostIds(i)
+    if (postsInFeed.toNumber() === 0) continue
+
+    const feedPosts = await legacyContract.getFeedPosts(
+      i,
+      0,
+      postsInFeed.toNumber()
+    )
+    console.log(`Feed ID: ${i}`)
+    feedPosts.forEach(async (post: OBSSStorage.PostStructOutput, j) => {
+      legacyPosts.push({
+        post: {
+          author: post.author,
+          metadata: {
+            digest: post.metadata.digest,
+            hashFunction: post.metadata.hashFunction,
+            size: post.metadata.size,
+          },
+          commentsFeedId: 0,
```

Comment:
Shouldn't it use `post.feedId` or `post.commentsFeedId`?

Reply:
`commentsFeedId` hardcoded to zero to pass input validations. It's going to generate a new `commentsFeedId` on adding a feed in contract, like this
`uint256 commentsFeedId = addFeed(legacyPost.post.metadata);`